lib/connection/transaction.ex

```diff
@@ -118,9 +124,12 @@ defmodule Janus.Connection.Transaction do
     transaction_status = transaction_status(pending_calls_table, transaction)
 
     case transaction_status do
-      {:ok, from} ->
+      {:ok, {from, type}} ->
         GenServer.reply(from, response)
-        :ets.delete(pending_calls_table, transaction)
+
+        if should_delete?(response, type) do
+          :ets.delete(pending_calls_table, transaction)
```

Comment:
Actual deletion should be logged

Reply:
We can change the way logs work here - please log from every case, this way without additional log lines we can include info about what happened (e.g. something was or wasn't removed from ets table)